src/main/java/bio/terra/cli/businessobject/resource/DataSource.java

```diff
@@ -0,0 +1,87 @@
+package bio.terra.cli.businessobject.resource;
+
+import bio.terra.cli.businessobject.Context;
+import bio.terra.cli.businessobject.Resource;
+import bio.terra.cli.businessobject.Workspace;
+import bio.terra.cli.serialization.persisted.resource.PDDataSource;
+import bio.terra.cli.serialization.userfacing.resource.UFDataSource;
+import bio.terra.cli.service.WorkspaceManagerService;
+import bio.terra.workspace.model.ResourceDescription;
+import bio.terra.workspace.model.TerraWorkspaceResource;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Internal representation of a data source referenced resource. Instances of this class are part of
+ * the current context or state.
+ */
+public class DataSource extends Resource {
+  private static final Logger logger = LoggerFactory.getLogger(DataSource.class);
+
+  public static final String SHORT_DESCRIPTION_KEY = "terra-short-description";
+  public static final String VERSION_KEY = "terra-version";
```

Comment:
i think terra-workspace-version and terra-workspace-short-description is clearer. is this field already set or can we change it still?